test/oauth-server/service/getUserInfo.go

```diff
@@ -6,5 +6,8 @@ func (requestHandler *RequestHandler) getUserInfo(w http.ResponseWriter, r *http
 	resp := map[string]interface{}{
 		"sub": deviceUserID,
 	}
-	jsonResponseWriter(w, resp)
+	err := jsonResponseWriter(w, resp)
+	if err != nil {
+		writeError(w, err, http.StatusBadRequest)
```

Comment:
when w.Write fails you can only log error. it is not recommended to call twice w.WriteHeader